.github/workflows/run-e2e-tests.yml

```diff
@@ -0,0 +1,63 @@
+name: Run E2E Test
+
+on:
+  push:
+    branches: [ "main" ]
+
+  pull_request:
+    branches: [ "main" ]
+    
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [20.x]
+
+    steps:
+    - name: Check Out Repository Code
+      uses: actions/checkout@v4
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        
+    - name: Install Foundry
+      uses: foundry-rs/foundry-toolchain@v1
+
+    - name: Fork Sepolia
+      run: |
+        anvil --fork-url https://sepolia.infura.io/v3/2f7bda55539a457686c696e9b3be21f1 &
+
+    - name: Install Dependencies
+      run: |
+        npm install
+
+    - name: 'Create env file'
+      run: |
+        rm -f .env
+        touch .env
+        echo WALLET_PRIVATE_KEY_A = ${{ secrets.WALLET_PRIVATE_KEY_A }} >> .env
+        echo WALLET_PRIVATE_KEY_B = ${{ secrets.WALLET_PRIVATE_KEY_B }} >> .env
+        echo WALLET_PRIVATE_KEY_C = ${{ secrets.WALLET_PRIVATE_KEY_C }} >> .env
+        echo SEPOLIA_MINTING_FEE_TOKEN = ${{ secrets.SEPOLIA_MINTING_FEE_TOKEN }} >> .env
+        echo SEPOLIA_RPC_PROVIDER_URL = http://localhost:8545 >> .env
+        echo SEPOLIA_MOCK_ERC721_ADDRESS = ${{ vars.SEPOLIA_NFT_CONTRACT_ADDRESS }} >> .env
+        echo SEPOLIA_LICENSING_MODULE_ADDRESS = ${{ vars.SEPOLIA_LICENSING_MODULE_ADDRESS }} >> .env
+        echo SEPOLIA_ROYALTY_POLICY_LAP_ADDRESS = ${{ vars.SEPOLIA_ROYALTY_POLICY_LAP_ADDRESS }} >> .env
+        echo SEPOLIA_DISPUTE_MODULE_ADDRESS = ${{ vars.SEPOLIA_DISPUTE_MODULE_ADDRESS }} >> .env
+        echo SEPOLIA_IPASSET_REGISTRY_ADDRESS = ${{ vars.SEPOLIA_IPASSET_REGISTRY_ADDRESS }} >> .env
+      
+    - name: Run E2E Test in Sepolia
+      run: |
+        npm run test:sepolia:e2e
+    - name: Upload Test Report
+      uses: actions/upload-artifact@v4
+      with:
+        name: sdk-sepolia-test-report
+        path: |
+          ./mochawesome-report
```

Comment:
It's better to add `if: always()` at the end of the "Upload Test Report" step, no matter what the e2e test is exited for any reason, the report will always be uploaded.